src/renderer/store/mutations.js

```diff
@@ -488,6 +489,33 @@ function setTradeHistory(state, trades) {
 
 function setOrderHistory(state, orders) {
   state.orderHistory = orders;
+
+  const orderHistoryStorageKey
+    = `orderhistory.${state.currentWallet.address}.${state.currentNetwork.net}.${assets.DEX_SCRIPT_HASH}`;
+  db.upsert(orderHistoryStorageKey, JSON.stringify(state.orderHistory));
+}
+function addToOrderHistory(state, newOrders) {
+  if (!state.orderHistory) {
+    state.orderHistory = [];
+  }
+
+  for (let i = 0; i < newOrders.length; i += 1) {
+    const existingOrderIndex = _.findIndex(state.orderHistory, (o) => {
+      return o.orderId === newOrders[i].orderId;
+    });
+
+    if (existingOrderIndex > -1) {
+      // this order is already in our cache, must be an update
+      // remove the existing order and add the updated version to the top
+      state.orderHistory.splice(existingOrderIndex, 1);
+    }
+
+    state.orderHistory.unshift(newOrders[i]);
+  }
+
+  const orderHistoryStorageKey
+    = `orderhistory.${state.currentWallet.address}.${state.currentNetwork.net}.${assets.DEX_SCRIPT_HASH}`;
+  db.upsert(orderHistoryStorageKey, JSON.stringify(state.orderHistory));
```

Comment:
Hmm, I just realized this is storing all order history in a single key in pouchdb; if we are going to store order history in pouchdb I think we need to store in a way that fetching new orders only has to write the new keys.

Reply:
removed caching in pouchdb